accepted/2025/cli-acquisition-tool.md

```diff
@@ -0,0 +1,224 @@
+She clones the repository, but the repo uses a global.json to enforce a certain tooling version for all contributors be installed to a repo-local directory.
+Since Mei doesn't have that version, she runs `dnup install` (or her editor does so on her behalf) and that version is downloaded and installed into her local repository copy.
+After this install is done, the next time she launches her editor it detects the install and she begins work on the feature she's selected.
+
+### Updating all installed toolchains
+
+Gavin has a few different SDKs managed via `dnup install` on their machine in the following configuration:
+
+- 8.0.404 - pinned by a `global.json` in `~/repos/mylib`
+- 9.0.303 - installed by tracking the 'latest' channel from `dnup install`
+- 10.0.100-preview.5 - installed tracking the 'preview' channel from `dnup install`
+
+They want to make sure their machine has all of the latest releases, so they run `dnup update`, which discovers:
+
+- there is a 8.0.405 version available
+- there is a 9.0.304 version available
+- there is a 10.0.100-preview.6 version available
+
+Since 8.0.404 is pinned by a specific repo, that version is not updated/removed, but 9.0.303 and 10.0.100-preview.5 are updated to 9.0.304 and 10.0.100-preview.6 respectively.
+
+## Requirements
+
+### Goals
+
+- **Unified SDK Management**: Provide a single tool for acquiring, updating, and managing .NET SDKs across all supported platforms
+- **User-local Installation**: Focus on user-writable locations that can be managed without elevation after initial setup
+- **New User Onboarding**: Streamline the getting-started experience, especially for users coming from Node.js or Rust ecosystems
+- **Version Management**: Support multiple concurrent SDK versions with ability to pin specific versions for projects
+- **Ease of updates**: Notify users when updates are available, and make it easy to get those updates
+- **Awareness of vulnerabilities**: Surface information on support status, vulnerabilities/CVEs, and other data to help users make an update decision
+- **IDE Integration**: Enable seamless integration with VS Code, Visual Studio, and other development environments
+- **Cross-platform Consistency**: Provide consistent experience across Windows, macOS, and Linux
+- **global.json Support**: Detect and respect project-specific SDK version requirements
+- **Self-maintenance**: Keep the bootstrapper tool itself up to date
+
+### Non-Goals
+
+- **Package Manager Integration/Global Installations**: Will not support installers beyond .zip/.tar installs initially
+- **Replace dotnet.exe**: The bootstrapper will not be dotnet.exe itself
+- **NuGet/Tools Support**: Tools are not in scope for `dnup` - instead they would belong to a NAOT `dnx` or NAOT'd `dotnet` CLI
+- **SDK Size Reduction**: Size reduction will be tackled separately, focused on deduplication and factoring out slices of functionality
+
+## Stakeholders and Reviewers
```

Comment:
Do you consider the Aspire team to be a stakeholder?

Reply:
At the time of this writing they weren't on the radar, but they definitely are now.